clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -61,22 +67,32 @@ public Builder(String groupId, List<TopicPartition> partitions) {
 
             this.data = new OffsetFetchRequestData()
                             .setGroupId(groupId)
+                            .setRequireStable(requireStable)
                             .setTopics(topics);
         }
 
-        public static Builder allTopicPartitions(String groupId) {
-            return new Builder(groupId, null);
-        }
-
-        public boolean isAllTopicPartitions() {
+        boolean isAllTopicPartitions() {
             return this.data.topics() == ALL_TOPIC_PARTITIONS;
         }
 
         @Override
         public OffsetFetchRequest build(short version) {
-            if (isAllTopicPartitions() && version < 2)
+            if (isAllTopicPartitions() && version < 2) {
                 throw new UnsupportedVersionException("The broker only supports OffsetFetchRequest " +
-                        "v" + version + ", but we need v2 or newer to request all topic partitions.");
+                    "v" + version + ", but we need v2 or newer to request all topic partitions.");
+            }
+
+            if (data.requireStable() && version < 7) {
+                log.info("Fallback the requireStable flag to false as broker " +
+                             "only supports OffsetFetchRequest version {}. Need " +
+                             "v7 or newer to enable this feature", version);
+
+                return new OffsetFetchRequest(new OffsetFetchRequestData()
```

Comment:
nit: do we need to create a new `OffsetFetchRequestData`?

Reply:
Actually we don't, I was thinking to maintain a original copy of data, but now this may introduce performance regression